pyprocessing/math.py

```diff
@@ -61,97 +72,103 @@ def _class_z():
+
+        raise AttributeError(f'PVector does not contain attribute {attr}')
+
+    def __setattr__(self, attr, val):
+        if SWIZZLE_RE.match(attr):
+            return self.__swizzle_set(attr, val)
+
+        return super().__setattr__(attr, val)
 
     def _instance_add(self, *vector, x=0, y=0, z=0):
-        if vector:
-            if isinstance(vector[0], PVector):
-                x, y, z = vector[0].position
-            elif isinstance(vector[0], (list, tuple)):
-                x, y, *_ = vector[0]
-                if len(vector[0]) == 3:
-                    z = vector[0][2]
-                else:
-                    z = None
-            elif all(isinstance(p, (int, float)) for p in vector):
-                x, y, z = vector
-            elif isinstance(vector[0], complex):
-                x, y, z = vector[0].real, vector[0].imag, None
-        vec = self.copy()
-        vec.x += x
-        vec.y += y
-        if z is not None:
-            vec.z += z
-        return vec
+        if len(vector) == 0:
+            vector = (x, y, z)
+
+        elif isinstance(vector[0], (list, tuple, PVector)):
+            vector = vector[0]
+
+        return self + vector
 
     def _instance_sub(self, *vector, x=0, y=0, z=0):
-        if vector:
-            if isinstance(vector[0], PVector):
-                x, y, z = vector[0].position
-            elif isinstance(vector[0], (list, tuple)):
-                x, y, *_ = vector[0]
-                if len(vector[0]) == 3:
-                    z = vector[0][2]
-                else:
-                    z = None
-            elif all(isinstance(p, (int, float)) for p in vector):
-                x, y, z = vector
-            elif isinstance(vector[0], complex):
-                x, y, z = vector[0].real, vector[0].imag, None
-        vec = self.copy()
-        vec.x -= x
-        vec.y -= y
-        if z is not None:
-            vec.z -= z
-        return vec
+        if len(vector) == 0:
+            vector = (x, y, z)
+
+        elif isinstance(vector[0], (list, tuple, PVector)):
+            vector = vector[0]
+
+        return self - vector
 
     def _instance_dist(self, v2):
-        return math.dist(self.position, v2.position)
+        return (self - v2).mag()
 
     def mult(self, scalar=1):
-        vec = self.copy()
-        vec.x *= scalar
-        vec.y *= scalar
-        if vec.z is not None:
-            vec.z *= scalar
-        return vec
+        return self * scalar
 
     def div(self, scalar=1):
-        vec = self.copy()
-        vec.x /= scalar
-        vec.y /= scalar
-        if vec.z is not None:
-            vec.z /= scalar
-        return vec
+        return self / scalar
 
-    @classmethod
-    def set_mag(cls, mag, target=None):
-        if target is not None:
-            target.normalize()
-            target *= mag
-            return target
-        cls.normalize()
-        cls *= mag
-        return cls
+    def set_mag(self, mag, target=None):
```

Comment:
Processing has a target argument for setMag, which sets the magnitude for that specific target. This behavior is lost here, please fix